### PR TITLE
Back button was added to composing message screen.

### PR DESCRIPTION
**Pull request by sanovikov71**
#2238 Change-Id: I1a6940991ff6ccffac13da4f276892f815d6a572

### CODE REVIEW

**Comment by sanovikov71** [comment]
#2238

**Comment by philipwhiuk** [comment]
I'm not sure I agree that just because you see a modal asking about saving the draft you don't then want to back to the Inbox.

I think we want to retain the fact they wanted to go 'up' and then use that instead of always passing in `false` when we are handling the dialog resolution.

**Comment by sanovikov71** [comment]
I uploaded another commit that works for dialogs too.

File: k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java [hunk 1]
```diff
@@ -1002,11 +1021,23 @@ public void onBackPressed() {
...
             }
         }
     }
 
+    private void openInboxFolder() {
+        String folder = account.getAutoExpandFolderName();
+        LocalSearch search = new LocalSearch(folder);
+        search.addAccountUuid(account.getUuid());
+        search.addAllowedFolder(folder);
+        MessageList.actionDisplaySearch(this, search, false, false);
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java]: This activity can be launched by third-party apps into their own task. This means you need to use `true` for the last argument so `MessageList` is launched into K-9 Mail's task. And then you'll need to `finish()` this activity.

File: k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java [hunk 2]
```diff
@@ -924,6 +934,10 @@ private void askBeforeDiscard() {
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
         switch (item.getItemId()) {
+            case android.R.id.home:
+                navigateUp = true;
+                prepareToFinish();
```
Comment (sanovikov71) [inline k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java]: done

File: k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java [hunk 3]
```diff
@@ -1002,11 +1021,23 @@ public void onBackPressed() {
...
+                if (navigateUp) {
+                    openInboxFolder();
+                } else {
+                    super.onBackPressed();
+                }
             }
         }
     }
 
+    private void openInboxFolder() {
```
Comment (sanovikov71) [inline k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java]: done

File: k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java (hunk 1 above)
Comment (sanovikov71) [inline k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java]: done

File: k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java [hunk 4]
```diff
@@ -1587,6 +1618,14 @@ public void run() {
         }
     };
 
+    private void initializeActionBar() {
+        ActionBar actionBar = getActionBar();
+
+        actionBar.setDisplayShowCustomEnabled(true);
+        actionBar.setCustomView(R.layout.actionbar_custom);
```
Comment (sanovikov71) [inline k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java]: done

**Comment by sanovikov71** [comment]
The comments are fixed.

**Comment by cketti** [comment]
Thanks!

I manually rebased/squased and merged.
https://github.com/k9mail/k-9/compare/05df114eb1ffeb9fc7960bb6afc48f0b38e30faf...d3aa9751ffeb018302ab447fd946705436b0a44d

**Pull request closed without merging**